### PR TITLE
go-bindings: Add ScanEventsMap and ScanEventsList

### PR DESCRIPTION
**Pull request by trosenblatt**
https://datadoghq.atlassian.net/browse/SDS-221

### CODE REVIEW

File: sds-go/go/scanner.go [hunk 1]
```diff
@@ -134,14 +125,94 @@ func (s *Scanner) Scan(event []byte) ([]byte, []RuleMatch, error) {
 	return processed, ruleMatches, nil
 }
 
+// Scan sends the string event to the SDS shared library for processing.
+// Returned values:
+//   - the processed log if any mutation happened
+//   - rules matches if any rule has matched
+//   - a possible error
+//
+// TODO(remy): implement ScanEventsMap, ScanEventsList
```
Comment (fuchsnj) [inline sds-go/go/scanner.go:134]: Should this TODO be removed now?

File: sds-go/go/scanner.go [hunk 2]
```diff
@@ -134,14 +125,94 @@ func (s *Scanner) Scan(event []byte) ([]byte, []RuleMatch, error) {
...
+	encodedEvent := make([]byte, 0)
+	encodedEvent, err := encodeListEvent(event, encodedEvent)
+	if err != nil {
+		return nil, nil, err
+	}
+	return s.scanEncodedEvent(encodedEvent)
+}
+
 // encodeStringEvent encodes teh given event to send it to the SDS shared library.
 // TODO(remy): implement encodeMapEvent, encodeListEvent
```
Comment (fuchsnj) [inline sds-go/go/scanner.go:163]: ... and this one?

File: sds-go/go/scanner.go (hunk 1 above)
Comment (trosenblatt) [reply sds-go/go/scanner.go]: fixed thks

File: sds-go/go/scanner.go (hunk 2 above)
Comment (trosenblatt) [reply sds-go/go/scanner.go]: fixed thks

File: sds-go/go/scanner.go [hunk 3]
```diff
@@ -134,14 +125,91 @@ func (s *Scanner) Scan(event []byte) ([]byte, []RuleMatch, error) {
...
+func (s *Scanner) Scan(event []byte) ([]byte, []RuleMatch, error) {
+	encodedEvent := make([]byte, 0)
+	encodedEvent, err := encodeStringEvent(event, encodedEvent)
+	if err != nil {
+		return nil, nil, err
+	}
+	return s.scanEncodedEvent(encodedEvent)
+}
+
+func (s *Scanner) ScanEventsMap(event map[string]interface{}) ([]byte, []RuleMatch, error) {
```
[the remark this answers is not in the document]
Comment (trosenblatt) [reply sds-go/go/scanner.go:147]: done thks. Yes I will add linter after that one

File: sds-go/go/scanner.go [hunk 4]
```diff
@@ -134,14 +125,91 @@ func (s *Scanner) Scan(event []byte) ([]byte, []RuleMatch, error) {
...
+	encodedEvent := make([]byte, 0)
+	encodedEvent, err := encodeMapEvent(event, encodedEvent)
+	if err != nil {
+		return nil, nil, err
+	}
+	return s.scanEncodedEvent(encodedEvent)
+}
+
+func (s *Scanner) ScanEventsList(event []interface{}) ([]byte, []RuleMatch, error) {
```
[the remark this answers is not in the document]
Comment (trosenblatt) [reply sds-go/go/scanner.go:161]: done thks

**Pull request merged**